src/relax/transform/fold_constant.cc

```diff
@@ -127,6 +129,19 @@ class ConstantFolder : public ExprMutator {
     return build_func;
   }
 
+  /*!
+   * \brief Checks if it is useful to fold \p expr.
+   * \details Folding an expr is a trade-off - we are materializing a constant in the IRModule and
+   * paying compile time cost to avoid the cost of executing this expr at runtime. For example,
+   * folding iota ops could result in large constants being materialized, thus increasing the size
+   * of the program.
+   */
+  bool ShouldBeFolded(Expr expr) {
+    // TODO(prakalp): Implement a heuristic to check if folding this op is actually useful or
```

Comment:
Would this include whether this `expr` is eligible or not? 
Or do we have a check before this heuristic and every `expr` at this point is eligible?

Reply:
This would not check if the expression is eligible, but it need not check that too. It would probably only need to check expr input and output shapes and decide if the folding would be useful. 

Another place to do this could be post matching call_tir and figured out its constant input args. We could then pass input/output shapes to figure out if folding would be useful.

Reply:
I’m curious about the plan of the heuristics here. As this check is standalone, it seems to me that we can introduce this function together with followup PRs.

Reply:
I introduced it here so we don't forget about it. Indeed there are other ways to achieve the same, perhaps a github issue. Is that preferred?